src/main/java/com/minecolonies/entity/ai/citizen/lumberjack/Tree.java

```diff
@@ -290,21 +358,36 @@ public void fillTreeStumps(@NotNull World world, int yLevel)
     }
 
     /**
-     * Adds the baseLog of the tree
+     * Adds the baseLog of the tree.
      */
     public void addBaseLog()
     {
         woodBlocks.add(new BlockPos(location));
     }
 
     /**
-     * Adds a log and searches for further logs(Breadth first search)
+     * Adds a log and searches for further logs(Breadth first search).
      *
-     * @param world The world the log is in
-     * @param log   the log to add
+     * @param world The world the log is in.
+     * @param log   the log to add.
      */
     private void addAndSearch(@NotNull World world, @NotNull BlockPos log)
     {
+        if(woodBlocks.size() >= MAX_TREE_SIZE)
+        {
+            return;
+        }
+
+        if(log.getY() < location.getY())
```

Comment:
Whats the point of even checking for getTopLog and getBaseLog if we are going to redo the check here. And why is this needed?